GETTING_STARTED.md

```diff
@@ -0,0 +1,192 @@
+
+# Getting Started
+
+This document is intended as an introduction to developing the Silver
+system.  It is *not* intended to introduce you to the concepts of
+extensible languages or using Silver, only the information you need
+for making changes to Silver itself.
+
+
+For anyone adding to this document:  
+Please keep this short.  It is just an introduction for anyone
+starting out, and we want them to actually read it.  Any questions
+which are too specific to be of general interest can be addressed to
+senior students.
+
+
+
+
+## Meta Level and Object Level
+
+One of the difficulties of developing the Silver system is keeping
+straight the meta level, the Silver compiler you are working on, and
+the object level, the Silver code you are compiling.  For example, if
+you are modifying the production for addition, you are writing code
+that works on `Expr`s with an attribute called `typerep` with a value
+of `intType()` at the meta level; your code is not manipulating
+integers, which are at the object level.
+
+
+As you spend more time working on Silver, you will grow more
+accustomed to the difference between the meta and object levels.  When
+starting out, you might occasionally need to slow down and think
+carefully to ensure you are thinking at the meta level as you work.
+
+
+
+
+## Scripts for Logistics
+
+There are several scripts for building Silver and other logistical
+operations in the `silver` repository.
+
+
+### `deep-rebuild`
+
+The `deep-rebuild` script does several rebuilds of Silver to ensure
+that any changes which have been made still permit Silver to build
+itself.
+
+
+
+### `self-compile`
+
+The `self-compile` script does a simple build of Silver to create a
```

Comment:
note that this doesn't write to jars, but sticks in in build; `cp build/silver.compiler.composed.Default.jar jars/silver.compiler.composed.Default.jar` may be necessary